pkg/service/model_build_target_group.go

```diff
@@ -221,62 +282,56 @@ func (t *defaultModelBuildTask) buildTargetGroupHealthCheckProtocol(_ context.Co
 	}
 }
 
-// buildTargetGroupPort constructs the TargetGroup's port.
-// Note: TargetGroup's port is not in the data path as we always register targets with port specified.
-// so this settings don't really matter to our controller, and we do our best to use the most appropriate port as targetGroup's port to avoid UX confusing.
-func (t *defaultModelBuildTask) buildTargetGroupPort(_ context.Context, targetType elbv2model.TargetType, svcPort corev1.ServicePort) int64 {
-	if targetType == elbv2model.TargetTypeInstance {
-		return int64(svcPort.NodePort)
-	}
-	if svcPort.TargetPort.Type == intstr.Int {
-		return int64(svcPort.TargetPort.IntValue())
-	}
-
-	// when a literal targetPort is used, we just use a fixed 1 here as this setting is not in the data path.
-	// also, under extreme edge case, it can actually be different ports for different pods.
-	return 1
-}
-
-func (t *defaultModelBuildTask) buildTargetGroupHealthCheckPath(_ context.Context) *string {
-	healthCheckPath := t.defaultHealthCheckPath
+func (t *defaultModelBuildTask) buildTargetGroupHealthCheckPath(_ context.Context, defaultHealthCheckPath string) *string {
```

Comment:
personally i think we should return string instead. since it always going to be a string instead of nil.

Reply:
I will change it in a follow up commit.